lib/shopify_transporter/pipeline/magento/order/line_items.rb

```diff
@@ -52,6 +57,42 @@ def tax_lines(item)
             ]
           end
 
+          def total_discount(item)
+            item['discount_amount'] if item['discount_amount'].present? && item['discount_amount'].to_f != 0
+          end
+
+          def fulfillment_status(item)
+            qty_ordered = qty_by_status(item, 'ordered')
+            qty_shipped = qty_by_status(item, 'shipped')
+            qty_refunded = qty_by_status(item, 'refunded')
+            if qty_ordered == qty_shipped && qty_refunded == 0
+              'fulfilled'
+            elsif qty_shipped > 0 && qty_shipped < qty_ordered
+              'partial'
+            end
+          end
+
+          def fulfillable_quantity(item)
+            qty_ordered = qty_by_status(item, 'ordered')
+            qty_shipped = qty_by_status(item, 'shipped')
+            qty_refunded = qty_by_status(item, 'refunded')
+
+            qty_ordered - [qty_shipped, qty_refunded].max
+          end
+
+          def requires_shipping?(item)
+            item['is_virtual'].to_i == 0 ? true : false
```

Comment:
In general, any predicate

```
X ? true : false
```

will always evaluate to the same value as simply 

```
X
```

In this case, this means you can change this method body to simply be:

```
item['is_virtual'].to_i == 0
```

Reply:
Good point! I'll update the code.